ocaml/xapi/xapi_database_backup.ml

```diff
@@ -0,0 +1,357 @@
+module D=Debug.Make(struct let name="xapi_database_backup" end)
+open D
+
+open Http
+open Xmlrpc_client
+open Db_cache_types
+open Stdext
+open Threadext
+
+type stat =
+  { created :  int64
+  ; modified : int64
+  ; deleted :  int64
+  } [@@deriving rpc]
+
+type field =
+  { fldname : string
+  ; stat :    stat
+  ; value :   Schema.Value.t
+  } [@@deriving rpc]
+
+type row =
+  { objref : string
+  ; stat :   stat
+  ; fields :  field list
+  } [@@deriving rpc]
+
+type table =
+  { tblname : string
+  ; stat :    stat
+  ; rows :   row list
+  } [@@deriving rpc]
+
+type del_tables =
+  { key:   string
+  ; table: string
+  } [@@deriving rpc]
+
+type delta =
+  { fresh_token: int64
+  ; last_event_token: int64
+  ; tables :     table list
+  ; deletes :    del_tables list
+  ; counts : (string * int) list
+  } [@@deriving rpc]
+
+type update =
+  { stat:stat
+  ; tblname:string
+  ; objref:string
+  ; fldname:string
+  ; value:Schema.Value.t
+  }
+
+exception Content_length_required
+let delta_to_string (d:delta) = Jsonrpc.to_string (rpc_of_delta d)
+
+let get_gen db =
+  db
+  |> Database.manifest
+  |> Manifest.generation
+
+(* Master Functions *)
+
+let get_table_list db =
+  TableSet.fold_over_recent (-2L) (fun table stat value acc -> table::acc) (Database.tableset db) []
```

Comment:
Just `TableSet.fold` would probably do here, since you aren't actually asking for the "recent" tables?